### PR TITLE
pod pkg: added multi net dual stack annotation func network.go

### PR DESCRIPTION
**Pull request by gkopels**
Add function to create annotations for multiple interfaces with dual stack IP addresses.

### CODE REVIEW

File: pkg/pod/network.go [hunk 1]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
 
 	return append(annotation, bond[0])
 }
+
+// StaticIPMultiNetAnnotation defines static ip address network annotation for pod object.
```
Comment (evgenLevin) [inline pkg/pod/network.go:87]: fix comment

File: pkg/pod/network.go [hunk 2]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
...
+func StaticIPMultiNetAnnotation(name string, ipAddr []string) *multus.NetworkSelectionElement {
+	return &multus.NetworkSelectionElement{
+		Name:      name,
+		IPRequest: ipAddr,
+	}
+}
+
+// StaticIPMultiNetDualStackAnnotation defines static ip address network annotation for pod object.
```
Comment (evgenLevin) [inline pkg/pod/network.go:95]: fix comment

File: pkg/pod/network.go [hunk 3]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
...
+func StaticIPMultiNetDualStackAnnotation(sriovNets, ipAddr []string) []*multus.NetworkSelectionElement {
+	annotation := []*multus.NetworkSelectionElement{}
+
+	// Verify ipAddr has an even number of IP addresses.
+	if len(ipAddr)%2 != 0 {
+		glog.V(100).Infof("ipAddr needs to contain an even number of IP addresses")
+		fmt.Println("ipAddr had an odd number of IP addresses")
```
Comment (evgenLevin) [inline pkg/pod/network.go:102]: remove

File: pkg/pod/network.go [hunk 4]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
...
+
+		return nil
```
Comment (evgenLevin) [inline pkg/pod/network.go:104]: why nil, not error?

File: pkg/pod/network.go [hunk 5]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
...
+	}
+
+	for index := 0; index < len(sriovNets); index++ {
```
Comment (evgenLevin) [inline pkg/pod/network.go:107]: use range

File: pkg/pod/network.go [hunk 6]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
...
+		sriovNetName := sriovNets[index]
+
+		// Verify that each net interfaces had 2 IP addresses.
+		if index*2+1 < len(ipAddr) {
```
Comment (evgenLevin) [inline pkg/pod/network.go:111]: if not?

File: pkg/pod/network.go [hunk 7]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
...
+		IPRequest: ipAddr,
+	}
+}
+
+// StaticIPMultiNetDualStackAnnotation defines static ip address network annotation for pod object.
+func StaticIPMultiNetDualStackAnnotation(sriovNets, ipAddr []string) []*multus.NetworkSelectionElement {
+	annotation := []*multus.NetworkSelectionElement{}
+
+	// Verify ipAddr has an even number of IP addresses.
+	if len(ipAddr)%2 != 0 {
```
Comment (evgenLevin) [inline pkg/pod/network.go:100]: Probably we should add more validations: sriovNets, ipAddr []string are not empty

File: pkg/pod/network.go (hunk 1 above)
Comment (gkopels) [reply pkg/pod/network.go:87]: Updated comment

File: pkg/pod/network.go (hunk 2 above)
Comment (gkopels) [reply pkg/pod/network.go:95]: updated

File: pkg/pod/network.go (hunk 4 above)
Comment (gkopels) [reply pkg/pod/network.go:104]: I am only returning []*multus.NetworkSelectionElement - should I error?

File: pkg/pod/network.go [hunk 8]
```diff
@@ -80,3 +83,37 @@ func StaticIPBondAnnotationWithInterface(
 
 	return append(annotation, bond[0])
 }
+
+// StaticIPMultiNetAnnotation defines static ip address network annotation for pod object.
+func StaticIPMultiNetAnnotation(name string, ipAddr []string) *multus.NetworkSelectionElement {
```
Comment (kononovn) [inline pkg/pod/network.go:88]: Please remove this function. Use  StaticIPAnnotation instead

File: pkg/pod/network.go (hunk 5 above)
Comment (gkopels) [reply pkg/pod/network.go:107]: changed to range

File: pkg/pod/network.go (hunk 7 above)
Comment (gkopels) [reply pkg/pod/network.go:100]: checking that they are not empty

File: pkg/pod/network.go (hunk 8 above)
Comment (gkopels) [reply pkg/pod/network.go:88]: This one is a [] and not sure how to use it in the for func
I will ping you

Comment (gkopels) [reply pkg/pod/network.go]: changed to use StaticIPAnnotation

File: pkg/pod/network.go (hunk 3 above)
Comment (gkopels) [reply pkg/pod/network.go]: removed

File: pkg/pod/network.go (hunk 4 above)
Comment (gkopels) [reply pkg/pod/network.go]: removed

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**